### PR TITLE
Switching to CSI 0.3.0

### PR DESCRIPTION
**Pull request by sbezverk**
Updating CSI vendor to use version 0.3.0

### CODE REVIEW

File: mock/service/controller.go [hunk 1]
```diff
@@ -341,3 +341,23 @@ func (s *service) ControllerGetCapabilities(
...
+	req *csi.DeleteSnapshotRequest) (*csi.DeleteSnapshotResponse, error) {
+	return &csi.DeleteSnapshotResponse{}, nil
+}
+
+func (s *service) ListSnapshots(ctx context.Context,
+	req *csi.ListSnapshotsRequest) (*csi.ListSnapshotsResponse, error) {
+	return &csi.ListSnapshotsResponse{}, nil
+}
+
+func (s *service) NodeGetInfo(ctx context.Context,
```
Comment (saad-ali) [inline mock/service/controller.go]: `NodeGetInfo` should be part of node service not controller

File: mock/service/controller.go [hunk 2]
```diff
@@ -341,3 +341,23 @@ func (s *service) ControllerGetCapabilities(
 		},
 	}, nil
 }
+
+func (s *service) CreateSnapshot(ctx context.Context,
+	req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	return &csi.CreateSnapshotResponse{}, nil
```
Comment (saad-ali) [inline mock/service/controller.go]: Do you want to return an error for now ("not implemented") until we can return a real mock object (like `CreateVolume(...)` does above)?

Comment (saad-ali) [reply mock/service/controller.go]: Same for `DeleteSnapshot` and `ListSnapshot` below?

File: mock/service/controller.go [hunk 3]
```diff
@@ -341,3 +341,23 @@ func (s *service) ControllerGetCapabilities(
 		},
```
Comment (saad-ali) [inline mock/service/controller.go]: Why is `mock/mock` binary file checked in? Let's remove that.

File: mock/service/controller.go (hunk 2 above)
Comment (sbezverk) [reply mock/service/controller.go]: make sense, will do for both

File: mock/service/controller.go (hunk 3 above)
Comment (sbezverk) [reply mock/service/controller.go]: will remove

File: mock/service/controller.go (hunk 1 above)
Comment (sbezverk) [reply mock/service/controller.go]: copy and past thing, thanks for catching..

File: mock/service/node.go [hunk 4]
```diff
@@ -227,3 +227,8 @@ func (s *service) NodeGetCapabilities(
 		},
 	}, nil
 }
+
+func (s *service) NodeGetInfo(ctx context.Context,
+	req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
+	return &csi.NodeGetInfoResponse{}, nil
```
Comment (saad-ali) [inline mock/service/node.go]: We should either return a proper response or `Not Implemented` error.

Comment (saad-ali) [reply mock/service/node.go]: Might be easy to just implement it. Look at `NodeGetId`

**Comment by saad-ali** [comment]
LGTM

**Comment by saad-ali** [comment]
Merging

**Pull request merged**